packages/charts/react-charting/src/components/FunnelChart/FunnelChart.base.tsx

```diff
@@ -0,0 +1,428 @@
+    setRefSelected(null);
+  }
+
+  function _onLegendSelectionChange(
+    legendsSelected: string[],
+    event: React.MouseEvent<HTMLButtonElement>,
+    currentLegend?: ILegend,
+  ): void {
+    if (props.legendProps?.canSelectMultipleLegends) {
+      setSelectedLegends(legendsSelected);
+    } else {
+      setSelectedLegends(legendsSelected.slice(-1));
+    }
+
+    if (props.legendProps?.onChange) {
+      props.legendProps.onChange(legendsSelected, event, currentLegend);
+    }
+  }
+
+  const _onLegendSelectionChangeCallback = React.useCallback(_onLegendSelectionChange, [props.legendProps]);
+
+  function getHighlightedLegend(): string[] {
+    return selectedLegends.length > 0 ? selectedLegends : hoveredStage ? [hoveredStage] : [];
+  }
+
+  function legendHighlighted(legend: string): boolean {
+    return getHighlightedLegend().includes(legend);
+  }
+
+  function noLegendHighlighted(): boolean {
+    return getHighlightedLegend().length === 0;
+  }
+
+  function _getEventHandlerProps(
+    data: IFunnelChartDataPoint | { stage: string; subValue: { category: string; value: number; color: string } },
+  ) {
+    if ('subValue' in data) {
+      return {
+        culture: props.culture,
+        onMouseOver: (event: React.MouseEvent<SVGElement>) => _handleStackedHover(data.stage, data.subValue, event),
+        onMouseMove: (event: React.MouseEvent<SVGElement>) => _handleStackedHover(data.stage, data.subValue, event),
+        onMouseOut: () => _handleMouseOut(),
+      };
+    } else {
+      return {
+        culture: props.culture,
+        onMouseOver: (event: React.MouseEvent<SVGElement>) => _handleHover(data, event),
+        onMouseMove: (event: React.MouseEvent<SVGElement>) => _handleHover(data, event),
+        onMouseOut: () => _handleMouseOut(),
+      };
+    }
+  }
+
+  function _renderSegmentText({
+    show,
+    x,
+    y,
+    value,
+    textColor,
+  }: {
+    show: boolean;
+    x: number;
+    y: number;
+    value: number;
+    textColor: string;
+  }) {
+    if (!show) {return null;}
+
+    const textElement = (
+      <text x={isRTL ? funnelWidth - x : x} y={y} textAnchor="middle" alignmentBaseline="middle" fill={textColor}>
+        {formatToLocaleString(value.toString(), props.culture)}
+      </text>
+    );
+
+    if (isRTL) {
+      return <g transform={`scale(-1,1) translate(${-funnelWidth},0)`}>{textElement}</g>;
+    }
+    return textElement;
+  }
+
+  function _renderFunnelSegment({
+    key,
+    pathD,
+    fill,
+    opacity,
+    textProps,
+    data,
+  }: {
+    key: string | number;
+    pathD: string;
+    fill: string;
+    opacity: number;
+    textProps?: {
+      show: boolean;
+      x: number;
+      y: number;
+      value: number;
+    };
+    data: IFunnelChartDataPoint | { stage: string; subValue: { category: string; value: number; color: string } };
+  }) {
+    const eventHandlers = _getEventHandlerProps(data);
+    const textColor = getInvertedTextColor(fill, props.theme);
```

Comment:
There is 1 more condition for this inversion. The color contrast ratio should be >= 3